apps/prairielearn/src/tests/helperQuestionPreview.ts

```diff
@@ -315,3 +316,34 @@ export function testFileDownloads(
     });
   });
 }
+
+export function testElementClientFiles(
+  previewPageInfo: QuestionPreviewPageInfo,
+  customElement: QuestionInfo,
+) {
+  const locals: any = previewPageInfo;
+
+  describe('setting up the submission data', function () {
+    it('should succeed', function () {
+      locals.shouldHaveButtons = ['grade', 'save', 'newVariant'];
+      locals.question = customElement;
+    });
+  });
+
+  helperQuestion.getInstanceQuestion(locals);
+  describe('downloading course text file', function () {
+    let elemList;
+    it('should contain a script tag for course-element.js', function () {
+      elemList = locals.$('script[src*="course-element.js"]');
+      assert.lengthOf(elemList, 1);
+    });
+
+    it('should download a file with the contents of course-element.js', async function () {
+      const fileUrl = locals.siteUrl + elemList[0].attribs.src;
+      const response = await fetch(fileUrl);
+      assert.equal(response.status, 200);
+      const fileContents = await response.text();
+      assert(fileContents.includes('This text was added by a script.'));
+    });
```

Comment:
Nit: I view the declaration of variables outside of tests to be a bit of a smell. In general I think we should prefer "fatter" tests that might make more assertions.

```suggestion
    it('should download a file with the contents of course-element.js', async function () {
      const elemList = locals.$('script[src*="course-element.js"]');
      assert.lengthOf(elemList, 1);
      
      const fileUrl = locals.siteUrl + elemList[0].attribs.src;
      const response = await fetch(fileUrl);
      assert.equal(response.status, 200);
      const fileContents = await response.text();
      assert(fileContents.includes('This text was added by a script.'));
    });
```

Reply:
done